### PR TITLE
enh: added option to smooth LDOS grid with Gaussian smoothing from sisl

### PR DESCRIPTION
**Pull request by sofiasanz**
Added possibility to smooth the real space grid with Gaussian smoothing for LDOS plots.

### CODE REVIEW

**Comment by zerothi** [comment]
Just an idea, you could let `smooth` be a dict which contains the arguments for the function, then you don't need to think too much about additional parameters?

**Comment by sofiasanz** [comment]
> Just an idea, you could let `smooth` be a dict which contains the arguments for the function, then you don't need to think too much about additional parameters?

Yes I like it! You can see my implementation in commit [cf3d5ec](https://github.com/dipc-cc/hubbard/pull/123/commits/cf3d5ec6b60085804c223db87b413ea1864c90b9).

**Comment by sofiasanz** [comment]
I see! Yes that is much better :-)

File: hubbard/plot/spectrum.py [hunk 1]
```diff
@@ -216,11 +216,19 @@ def __init__(self, HH, wavefunction, sites=[], ext_geom=None, realspace=False, *
...
-                grid += slice_grid.real**2 + slice_grid.imag**2
+                grid_i.grid = np.absolute(grid_i.grid) ** 2
 
-            self.__realspace__(grid, **kwargs)
+                grid += grid_i.grid
+
+            grid_i.grid = grid
+
+            # In Ref. [ACS Nano, Vol. 12, No. 7, p. 7048-7056] they use r=0.6 Angstroms to simulate the Gaussian-shaped tip
+            grid_i = grid_i.smooth(**kwargs["smooth"])
```
Comment (zerothi) [inline hubbard/plot/spectrum.py:226]: I don't know if this what you want, but this will always smooth it?

**Comment by sofiasanz** [comment]
But if the user doesn't pass any arguments for `smooth` then it assumes `r=0`, no? And then it doesn't have the smoothing defect, no?

**Comment by sofiasanz** [comment]
Otherwise it can just first check if the `smooth` dictionary is empty or not.

**Comment by zerothi** [comment]
> But if the user doesn't pass any arguments for `smooth` then it assumes `r=0`, no? And then it doesn't have the smoothing effect, no?

No, it will use the default values, https://zerothi.github.io/sisl/api/generated/sisl.Grid.html?highlight=smooth#sisl.Grid.smooth

Remember that kwargs for a zero length dict is effectively passing no arguments. I think it is safest to check whether the argument exist (checking the length would not allow users to use default values).

**Comment by sofiasanz** [comment]
I see. Forgot to check which were the default values of the method. Thanks!

> Remember that kwargs for a zero length dict is effectively passing no arguments. I think it is safest to check whether the argument exist (checking the length would not allow users to use default values).

I just saw again what you wrote in https://github.com/dipc-cc/hubbard/pull/123#commitcomment-69471285, didn't see the first line before hehe

**Review by zerothi — APPROVED** [review]
Lgtm!

**Pull request merged**